From 9e205da9b7599135071b5690c5c96f380f7aec38 Mon Sep 17 00:00:00 2001
From: Allen Ye <allen.ye@mediatek.com>
Date: Thu, 28 Apr 2022 17:39:33 +0800
Subject: [PATCH] HAL: refactor wifi_setRadioOperatingChannelBandwidth()

---
 source/wifi/wifi_hal.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/source/wifi/wifi_hal.c b/source/wifi/wifi_hal.c
index 48e6021..84e2c22 100644
--- a/source/wifi/wifi_hal.c
+++ b/source/wifi/wifi_hal.c
@@ -2060,15 +2060,17 @@ INT wifi_setRadioOperatingChannelBandwidth(INT radioIndex, CHAR *output_string)
             pptr->value="1";
         else if(strcmp(output_string,"80MHz") == 0)
             pptr->value="1";
+        else 
+            pptr->value="0";
 
         pptr++; // added ieee80211n
 
-	if(strcmp(output_string,"80MHz") == 0)
-	{
-	    pptr->name="ieee80211ac";
-	    pptr->value="1";
-	    pptr++; // added ieee80211ac
-	}
+        pptr->name="ieee80211ac";
+        if(strcmp(output_string,"80MHz") == 0)
+            pptr->value="1";
+        else
+            pptr->value="0";
+        pptr++;
     }
 
     for(int i=0; i<=MAX_APS/NUMBER_OF_RADIOS; i++)
-- 
2.18.0

