diff --git a/source/dmltad/cosa_wanconnectivity_operations.c b/source/dmltad/cosa_wanconnectivity_operations.c
index 51dc1092..677f4a2d 100644
--- a/source/dmltad/cosa_wanconnectivity_operations.c
+++ b/source/dmltad/cosa_wanconnectivity_operations.c
@@ -1224,7 +1224,11 @@ int send_query(struct query *query_info,struct mk_query *query_list,BOOL use_raw
 
 wait_for_response:
         start_time = clock_mono_ms();
+#ifdef _64BIT_ARCH_SUPPORT_
+        WANCHK_LOG_DBG("waiting for response with timeout:%ld\n",timeout);
+#else
         WANCHK_LOG_DBG("waiting for response with timeout:%lld\n",timeout);
+#endif
         if (timeout >= 0) {
             rc = poll(&pfd, 1, timeout);
         } else {
