diff --git a/src/platform/platform_hal.c b/src/platform/platform_hal.c
index 845b80f..b02bb07 100644
--- a/src/platform/platform_hal.c
+++ b/src/platform/platform_hal.c
@@ -236,3 +236,22 @@ INT platform_hal_GetRouterRegion(CHAR* pValue)
 {
 	return RETURN_OK;
 }
+
+char *get_current_wan_ifname()
+{
+return "0";
+}
+
+
+
+INT platform_hal_GetDhcpv6_Options ( dhcp_opt_list ** req_opt_list, dhcp_opt_list ** send_opt_list)
+{
+return RETURN_OK;
+}
+
+
+
+INT platform_hal_GetDhcpv4_Options ( dhcp_opt_list ** req_opt_list, dhcp_opt_list ** send_opt_list)
+{
+return RETURN_OK;
+}
\ No newline at end of file
